libyul/AssemblyStack.cpp

```diff
@@ -185,7 +185,9 @@ void AssemblyStack::optimize(Object& _object, bool _isCreation)
 		meter.get(),
 		_object,
 		m_optimiserSettings.optimizeStackAllocation,
-		m_optimiserSettings.yulOptimiserSteps
+		m_optimiserSettings.yulOptimiserSteps,
+		{},
+		m_optimiserSettings.expectedExecutionsPerDeployment
```

Comment:
I wonder whether the optimizer context shouldn't just get the full optimizer settings... but can be done in the future, no need to change the PR because of it.
But more importantly: this should be
```suggestion
		_isCreation ? nullopt : m_optimiserSettings.expectedExecutionsPerDeployment
```
shouldn't it?